### DISCUSSION TITLE
请问hy2是否存在二次TLS加密的问题？

### DISCUSSION THREAD

**Discussion by yh317** [apernet/hysteria#1025] [Q&A]
请问HY2的加密机制是什么样的，是像trojan那样加密所有数据，还是像vless那样只加密包头？

**Comment by lry127** [comment]
像trojan一样，hy2（实现了[http3](https://datatracker.ietf.org/doc/rfc9114/)）会使用tls 1.3加密完整hy2报文，见 [协议声明](https://v2.hysteria.network/docs/developers/Protocol/). 但是代理的tcp连接（例如浏览器发送的https请求）本身可能使用了tls，所以在hy2 quic的基础上，内部报文可能是tls流量。无论如何，hy2可以认为是安全的。

**Comment by tobyxdd** [comment]
1. 所有数据用 TLS 1.3 加密，如果内层也是 TLS 那就是二次加密
2. vless 本身并不是只加密包头，vless 根本没有加密。你想说的大概是 XTLS Vision
